src/python/bot/tasks/trials.py

```diff
@@ -20,34 +20,37 @@
 from system import environment
 
 
-def setup_additional_args_for_app():
-  """Select additional args for the specified app at random."""
-  if environment.is_engine_fuzzer_job():
-    # Not applicable to engine fuzzers.
-    return
-
-  app_name = environment.get_value('APP_NAME')
-  if not app_name:
-    return
-
-  # Convert the app_name to lowercase. Case may vary by platform.
-  app_name = app_name.lower()
-
-  # Hack: strip file extensions that may be appended on various platforms.
-  extensions_to_strip = ['.exe', '.apk']
-  for extension in extensions_to_strip:
-    app_name = utils.strip_from_right(app_name, extension)
-
-  trials = data_types.Trial.query(data_types.Trial.app_name == app_name)
-  trials = [trial for trial in trials if random.random() < trial.probability]
-  if not trials:
-    return
-
-  app_args = environment.get_value('APP_ARGS', '') + ' ' + trials[0].app_args
-  trial_app_args = trials[0].app_args
-  for trial in trials[1:]:
-    app_args += ' ' + trial.app_args
-    trial_app_args += ' ' + trial.app_args
-
-  environment.set_value('APP_ARGS', app_args)
-  environment.set_value('TRIAL_APP_ARGS', trial_app_args)
+class Trials:
+  """Helper class for selecting app-specific extra flags."""
+
+  def __init__(self):
+    self.trials = []
+
+    app_name = environment.get_value('APP_NAME')
+    if not app_name:
+      return
+
+    # Convert the app_name to lowercase. Case may vary by platform.
+    app_name = app_name.lower()
+
+    # Hack: strip file extensions that may be appended on various platforms.
+    extensions_to_strip = ['.exe', '.apk']
+    for extension in extensions_to_strip:
+      app_name = utils.strip_from_right(app_name, extension)
+
+    self.trials = data_types.Trial.query(data_types.Trial.app_name == app_name)
```

Comment:
add list() around "data_types.Trial.query(data_types.Trial.app_name == app_name)" so that it computes the query result.

Reply:
Done